app/src/main/java/com/framgia/moneymanagement/screen/register/RegisterPresenter.java

```diff
@@ -0,0 +1,89 @@
+package com.framgia.moneymanagement.screen.register;
+
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
+import com.framgia.moneymanagement.data.model.User;
+import com.framgia.moneymanagement.data.repository.AuthenticationRepository;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthWeakPasswordException;
+
+public class RegisterPresenter implements RegisterContact.Presenter,
+        OnCompleteListener, OnFailureListener {
+    private AuthenticationRepository mAuthencateRepository;
+    private RegisterContact.View mView;
+
+    public RegisterPresenter(AuthenticationRepository authencateRepository, RegisterContact.View view) {
+        mAuthencateRepository = authencateRepository;
+        mView = view;
+    }
+
+    @Override
+    public void createAccount(String userName, String password, String confirmPassword) {
+        if (!password.equals(confirmPassword)) {
+            mView.onCreateFail();
```

Comment:
em cần show chính xác lỗi cho người dùng
tuơng ứng với các case có thể xảy ra
1 user name empty
2 password empty
3 confirm password empty
4. password and confirm does not match
5. login eror

Reply:
các lỗi này cần xử lý trong method validateData